### ISSUE TITLE
Linker warning with `ld` from binutils 2.42-4ubuntu2.6

### ISSUE THREAD

**Issue by PatrickHaecker** [JuliaLang/PackageCompiler.jl#1071]
PackageCompiler.jl works here on an Ubuntu 24.04 system, but I get a warning at the end
```
[…]
Precompiling packages finished.
  25 dependencies successfully precompiled in 32 seconds. 72 already precompiled.
⡆ [07m:12s] PackageCompiler: compiling incremental system image
/usr/bin/ld: warning: text#3.o: missing .note.GNU-stack section implies executable stack
/usr/bin/ld: NOTE: This behaviour is deprecated and will be removed in a future version of the linker
```

I am using GNU `ld` from package binutils 2.42-4ubuntu2.6.

**Comment by PatrickHaecker** [comment]
I am using Julia 1.12.1.

The warning is probably printed by Julia itself when the [cmd compiling the system image](https://github.com/JuliaLang/PackageCompiler.jl/blob/c1e37c44f591a35681437c2ff6fcb428b40ed528/src/PackageCompiler.jl#L541) is run.

According to the [release announcement from GNU ld from binutils 2.39](https://lists.gnu.org/archive/html/info-gnu/2022-08/msg00002.html) a `-z noexecstack` is needed to avoid this warning. [Julia uses this flag](https://github.com/JuliaLang/julia/blob/c0d34aa7ba29e2ba642f3ab158dc67f2be0fc19a/Make.inc#L1541), but, because the warning is still printed, most likely not in this situation.

So should I close this issue and open an issue in Julia instead?
